src/libcollections/string.rs

```diff
@@ -29,7 +36,185 @@ use str::{self, FromStr, Utf8Error, Chars};
 use vec::Vec;
 use boxed::Box;
 
-/// A growable string stored as a UTF-8 encoded buffer.
+//! A UTF-8 encoded, growable string.
+//!
+//! The `String` type is the most common string type that has ownership over the
+//! contents of the string. It has a close relationship with its borrowed
+//! counterpoint, the primitive [`str`].
+//!
+//! [`str`]: ../primitive.str.html
+//!
+//! # Examples
+//!
+//! You can create a `String` from a literal string with `String::from`:
+//!
+//! ```
+//! let hello = String::from("Hello, world!");
+//! ```
+//!
+//! You can append a [`char`] to a `String` with the [`push()`] method, and
+//! append a [`&str`] with the [`push_str()`] method:
+//!
+//! ```
+//! let mut hello = String::from("Hello, ");
+//!
+//! hello.push('w');
+//! hello.push_str("orld!");
+//! ```
+//!
+//! [`push()`]: #method.push
+//! [`push_str()`]: #method.push_str
+//!
+//! If you have a vector of UTF-8 bytes, you can create a `String` from it with
+//! the [`from_utf8()`] method:
+//!
+//! ```
+//! // some bytes, in a vector
+//! let sparkle_heart = vec![240, 159, 146, 150];
+//!
+//! // We know these bytes are valid, so we'll use `unwrap()`.
+//! let sparkle_heart = String::from_utf8(sparkle_heart).unwrap();
+//!
+//! assert_eq!("💖", sparkle_heart);
+//! ```
+//!
+//! [`from_utf8()`]: #method.from_utf8
+//!
+//! # UTF-8
+//!
+//! `String`s are always valid UTF-8. This has a few implications. The first of
```

Comment:
True, but the following sentence isn't really a complete sentence otherwise ("The first of which").